pkg/bot/lark.go

```diff
@@ -0,0 +1,225 @@
+	ClusterName      string
+	DefaultNamespace string
+	Port             int
+	MessagePath      string
+	LarkClient       *utils.LarkClient
+}
+
+// NewLarkBot returns new Bot object
+func NewLarkBot(c *config.Config) Bot {
+	larkConf := c.Communications.Lark
+	appSettings := core.NewInternalAppSettings(core.SetAppCredentials(larkConf.AppID, larkConf.AppSecret),
+		core.SetAppEventKey(larkConf.VerificationToken, larkConf.EncryptKey))
+	conf := core.NewConfig(core.Domain(larkConf.Endpoint), appSettings, core.SetLoggerLevel(core.LoggerLevelError))
+	return &LarkBot{
+		AllowKubectl:     c.Settings.Kubectl.Enabled,
+		RestrictAccess:   c.Settings.Kubectl.RestrictAccess,
+		ClusterName:      c.Settings.ClusterName,
+		DefaultNamespace: c.Settings.Kubectl.DefaultNamespace,
+		Port:             c.Communications.Lark.Port,
+		MessagePath:      c.Communications.Lark.MessagePath,
+		LarkClient:       utils.NewLarkClient(conf),
+	}
+}
+
+// Execute commands sent by users
+func (l *LarkBot) Execute(e map[string]interface{}) {
+	event, err := AccessAndTypeCastToMap(Event, e)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToMap: %s", err.Error())
+		return
+	}
+
+	chatType, err := AccessAndTypeCastToString(ChatType, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	text, err := AccessAndTypeCastToString(Text, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	executor := execute.NewDefaultExecutor(text, l.AllowKubectl, l.RestrictAccess, l.DefaultNamespace,
+		l.ClusterName, config.LarkBot, "", true)
+	response := executor.Execute()
+
+	if chatType == "group" {
+		chatID, err := AccessAndTypeCastToString(OpenChatID, event)
+		if err != nil {
+			log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+			return
+		}
+
+		l.LarkClient.SendTextMessage(ChatID, chatID, response)
+		return
+	}
+	openID, err := AccessAndTypeCastToString(OpenID, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+	l.LarkClient.SendTextMessage(OpenID, openID, response)
+}
+
+// Start starts the lark server and listens for lark messages
+func (l *LarkBot) Start() {
+	// See: https://open.larksuite.com/document/ukTMukTMukTM/ukjNxYjL5YTM24SO2EjN
+	// message
+	larkConf := l.LarkClient.Conf
+	event.SetTypeCallback(larkConf, Message, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.Execute(e)
+		return nil
+	})
+
+	// add_bot
+	event.SetTypeCallback(larkConf, AddBot, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	// p2p_chat_create
+	event.SetTypeCallback(larkConf, P2pChatCreate, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
```

Comment:
Same here. Can we ignore such events?

Reply:
@PrasadG193  It may not have been as clear as I explained earlier, but this lark event callback really can't be ignored. In the lark chat group, we need to respond when these things happen.